base/reducedim.jl

```diff
@@ -695,7 +699,7 @@ function findminmax!(f, Rval, Rind, A::AbstractArray{T,N}) where {T,N}
     keep, Idefault = Broadcast.shapeindexer(indsRt)
     ks = keys(A)
     y = iterate(ks)
-    zi = zero(eltype(ks))
+    zi = _get_unused_index(A)
```

Comment:
For safety, I think you want to make this `_get_unused_index(first(ks))` and accommodate the different types of keys. You can't promise that someone hasn't written a specialized `keys` function, and you'll note below that what gets stored in `Rind` is an entry in `ks`.